src/Compiler/Checking/infos.fs

```diff
@@ -808,6 +808,15 @@ type MethInfo =
         | ProvidedMeth(_, mi, _, m) -> [mi.PUntaint((fun mi -> mi.GetParameters().Length), m)] // Why is this a list? Answer: because the method might be curried
 #endif
 
+    /// Indicates if the property is a IsABC union case tester implied by a union case definition
+    member x.IsUnionCaseTester =
+        let tcref = x.ApparentEnclosingTyconRef
+        tcref.IsUnionTycon &&
+        x.LogicalName.StartsWith("get_Is") &&
```

Comment:
Is there something on top this can rely on, to prevent clashes without user-defined Is**** properties?

Reply:
Like the implied val check immediately below your comment? :)